src/lib.rs

```diff
@@ -83,20 +83,22 @@ pub unsafe extern "C" fn rakaly_melt_write_data(
     res: *const MeltedBuffer,
     buffer: *mut c_char,
     length: size_t,
-) -> c_int {
+) -> size_t {
     if res.is_null() || buffer.is_null() {
-        return -1;
+        let result = -1;
+        return result as size_t;
```

Comment:
Should we make the change in the error case to be 0 like `rakaly_melt_data_length`. Just realized maybe it's worth having the conformity here.

Reply:
Yeah, that's probably better.